### PR TITLE
add boundclause tostring

### PR DESCRIPTION
**Pull request by nevermore3**

### CODE REVIEW

File: src/parser/TraverseSentences.cpp [hunk 1]
```diff
@@ -222,4 +222,30 @@ std::string YieldSentence::toString() const {
...
+    buf.reserve(256);
+    buf += "GET SUBGRAPH ";
+    if (step_ != nullptr) {
+        buf += step_->toString();
+    }
+    if (from_ != nullptr) {
+        buf += " ";
+        buf += from_->toString();
+    }
+    if (in_ != nullptr) {
```
Comment (Shylock-Hg) [inline src/parser/TraverseSentences.cpp:237]: There is the `in` , `out` and `both` information already.

File: src/parser/Clauses.h [hunk 2]
```diff
@@ -357,8 +357,15 @@ class GroupClause final {
 
 class InBoundClause final {
 public:
-    explicit InBoundClause(OverEdges *edges) {
+    enum BoundType : uint8_t {
+        IN,
```
Comment (Shylock-Hg) [inline src/parser/Clauses.h:361]: It's extra.

File: src/parser/TraverseSentences.cpp (hunk 1 above)
Comment (nevermore3) [reply src/parser/TraverseSentences.cpp:237]: pls be more specific

File: src/parser/Clauses.h [hunk 3]
```diff
@@ -367,10 +374,12 @@ class InBoundClause final {
 
 private:
     std::unique_ptr<OverEdges> overEdges_;
+    BoundType boundType_;
 };
 
-using OutBoundClause = InBoundClause;
-using BothInOutClause = InBoundClause;
+using InBoundClause = BoundClause;
```
Comment (Shylock-Hg) [inline src/parser/Clauses.h:380]: InBoundClause with ctor to set `IN` as value of `boundType_`.

**Review by laura-ding — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**